builder/amazon/common/step_security_group.go

```diff
@@ -45,7 +45,9 @@ func (s *StepSecurityGroup) Run(state multistep.StateBag) multistep.StepAction {
 
 	port := s.CommConfig.Port()
 	if port == 0 {
-		panic("port must be set to a non-zero value.")
+		if s.CommConfig.Type != "none" {
+			panic("port must be set to a non-zero value.")
```

Comment:
yeah I think basically everything is supposed to happen like normal, sans connection.

the null communicator seems like it fits poorly into packer, so I'm not surprised things break further down the line. There may be more steps which must be elided.

It might take more thinking about how to more elegantly decouple the code from the communicators.